logtail/logtail.go

```diff
@@ -290,42 +298,67 @@ func (l *Logger) drainBlock() (shuttingDown bool) {
 
 // drainPending drains and encodes a batch of logs from the buffer for upload.
 // If no logs are available, drainPending blocks until logs are available.
-func (l *Logger) drainPending() (res []byte) {
-	buf := &l.drainBuf
-	buf.Reset()
-	buf.WriteByte('[')
-	entries := 0
+// The returned buffer is only valid until the next call to drainPending.
+func (l *Logger) drainPending() (b []byte) {
+	b = l.drainBuf[:0]
+	b = append(b, '[')
+	defer func() {
+		b = bytes.TrimRight(b, ",")
+		b = append(b, ']')
+		l.drainBuf = b
+		if len(b) <= len("[]") {
+			b = nil
+		}
+	}()
 
 	var batchDone bool
-	const maxLen = 256 << 10
-	for buf.Len() < maxLen && !batchDone {
```

Comment:
batchDone is always false, I think(?) you can just remove it.

Reply:
Done.